test/e2e/utils/kubernetes.go

```diff
@@ -549,3 +550,28 @@ func runOperatorLocally(stopCh chan struct{}, namespace string) {
 	_ = os.Setenv("KUBECONFIG", kubeConfig)
 	launcher.Launch(launcher.Parameters{StopChannel: stopCh})
 }
+func (k TestKubernetes) DeleteCache(cache *ispnv2.Cache) {
+	err := k.Kubernetes.Client.Delete(context.TODO(), cache)
+	ExpectMaybeNotFound(err)
+}
+
+func (k TestKubernetes) WaitForCacheCondition(name, namespace string, condition v2alpha1.CacheCondition) {
+	cache := &v2alpha1.Cache{}
+	err := wait.Poll(ConditionPollPeriod, ConditionWaitTimeout, func() (done bool, err error) {
+		err = k.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: name}, cache)
+		if err != nil && errors.IsNotFound(err) {
+			return false, err
+		}
+		if err != nil {
+			return false, nil
+		}
+		for _, c := range cache.Status.Conditions {
+			if (strings.ToLower(c.Type) == strings.ToLower(condition.Type)) && (c.Status == condition.Status) {
```

Comment:
Overall good, @durgeshanaokar 
It would be nice to move comparison and validation logic to the Cache CR type level, but probably not in this PR
Comparing with ignore case not needed for Cache CR conditions, as we used before some changes lowercase conditions values for the Infinispan CR and such kind comparison requires for migration purposes
Also in separate PR we will need to create a separate Cache CR condition type and enums the same as for Infinispan CR
Please run `make lint` before to validate linters warnings, starting #960 CI stops on any linter warning